pkg/fixtures/writer_s3.go

```diff
@@ -0,0 +1,108 @@
+package fixtures
+
+import (
+	"context"
+	"github.com/applike/gosoline/pkg/blob"
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/aws/aws-sdk-go/aws"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type BlobFixturesSettings struct {
+	BasePath   string
+	ConfigName string
+}
+
+type blobFixtureWriter struct {
+	logger      mon.Logger
+	batchRunner *blob.BatchRunner
+	purger      *blobPurger
+	settings    *BlobFixturesSettings
+	store       blob.Store
+}
+
+func BlobFixtureWriterFactory(settings *BlobFixturesSettings) FixtureWriterFactory {
+	return func(config cfg.Config, logger mon.Logger) (FixtureWriter, error) {
+		store := blob.NewStore(config, logger, settings.ConfigName)
+		br := blob.NewBatchRunner(config, logger)
+		purger := newBlobPurger(config, logger, settings)
+
+		return NewBlobFixtureWriterWithInterfaces(logger, store, br, settings, purger), nil
+	}
+}
+
+func NewBlobFixtureWriterWithInterfaces(logger mon.Logger, store blob.Store, batchRunner *blob.BatchRunner, settings *BlobFixturesSettings, purger *blobPurger) FixtureWriter {
```

Comment:
should we change the parameters into the order that is mentioned in our coding guidelines? :thinking:

Reply:
Please read the guidelines carefully again and check your complete changeset, if you adhere to it.

Reply:
Guideline doesn't say anything concerning constructors params sorting. that's maybe something to include.

Reply:
Applied